src/hooks/usePollTableQuery.ts

```diff
@@ -0,0 +1,119 @@
+import { useState, useEffect } from "react";
+import {
+  isNetworkRequestInFlight,
+  NetworkStatus,
+} from "apollo-client/core/networkStatus";
+import { usePrevious } from "hooks";
+import { useParams, useLocation } from "react-router-dom";
+import { ApolloQueryResult } from "apollo-client";
+import isEqual from "lodash.isequal";
+
+export const usePollTableQuery = <ApolloQueryVariables, ApolloQueryResultType>({
+  networkStatus,
+  getQueryVariables,
+  refetch,
+  search,
+}: {
+  networkStatus: NetworkStatus;
+  getQueryVariables: (search: string, id?: string) => ApolloQueryVariables;
+  refetch: (
+    variables?: ApolloQueryVariables
+  ) => Promise<ApolloQueryResult<ApolloQueryResultType>>;
+  search: string;
```

Comment:
when there is more than one or two and the types are long and take multiple generics, i think having a separate interface to represent the type creates cleaner, more readable code.